src/torchcodec/_core/AVIOTensorContext.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <torch/types.h>
+#include "src/torchcodec/_core/AVIOContextHolder.h"
+
+namespace facebook::torchcodec {
+
+struct TensorContext {
+  torch::Tensor data;
+  int64_t current;
+};
+
```

Comment:
I don't love that this is exposed outside of this file.

Reply:
Curious how we could prevent it from leaking out? A base class with a definition as `private`? That sounds overkill?